app/assets/javascripts/scraper.js

```diff
@@ -1,13 +1,24 @@
 (function() {
   function scrape(e) {
-    var query = $('.scraper-query').val();
-    var url = $(this).attr('href') + '?query=' + encodeURIComponent(query);
-    $('.scraper-results').load(url);
-    e.preventDefault();
+    var that = $(this);
+    if (!that.data('scrape')) {
+      return false;
+    }
+
+    delay(function() {
+      var query = that.val();
+      if (query.length < 3) {
+        return false;
+      }
+
+      var url = that.data('scrape-path') + '?query=' + encodeURIComponent(query);
+      Spinner.show('.scraper-results');
+      $('.scraper-results').load(url);
+    }, 500);
   }
 
   function ready() {
-    $('.scraper-link').on('click', scrape);
+    $('.scraper-query').on('keyup', scrape);
   }
 
   $(document).on('turbolinks:load', ready);
```

Comment:
Should we simplify `$(document).on('turbolinks:load', ready);` to something like this:

```javascript
 $(document).on('turbolinks:load', function(){
   $('.scraper-query').on('keyup', scrape);
 });
```

Currently, I do not see any advantage in having a separate ready function. Moreover, it would be cool to write 1-2 sentences about the purpose of this jquery event listener.